dbt/models/iasworld/schema/iasworld.dweldat.yml

```diff
@@ -1376,12 +1402,42 @@ sources:
             description: '{{ doc("shared_column_char_ncu") }}'
             tests:
               - not_null:
-                  name: iasworld_dweldat_char_ncu_not_null
-                  additional_select_columns: *select-columns
-                  config: *unique-conditions
+                  name: iasworld_dweldat_char_ncu_not_null_when_class_is_212
+                  additional_select_columns: *select-columns-with-class
+                  config: &unique-conditions-class-212
+                    where: |
+                      CAST(taxyr AS int) BETWEEN {{ var('test_qc_year_start') }} AND {{ var('test_qc_year_end') }}
+                      AND cur = 'Y'
+                      AND deactivat IS NULL
+                      AND class = '212'
                   meta:
                     description: >
                       user20 (No. of Commercial Units) should not be null
+                      when class is 212
+              - not_accepted_values:
+                  name: iasworld_dweldat_char_ncu_not_0_when_class_is_212
+                  values:
+                    - 0
+                  additional_select_columns: *select-columns-with-class
+                  config: *unique-conditions-class-212
+                  meta:
+                    description: >
+                      user20 (No. of Commercial Units) should not be 0
+                      when class is 212
+              - is_null:
+                  name: iasworld_dweldat_char_ncu_null_when_class_is_not_212_and_char_ncu_not_0
+                  additional_select_columns: *select-columns-with-class
+                  config:
+                    where: |
+                      CAST(taxyr AS int) BETWEEN {{ var('test_qc_year_start') }} AND {{ var('test_qc_year_end') }}
+                      AND cur = 'Y'
+                      AND deactivat IS NULL
+                      AND class != '212'
```

Comment:
These test failures include a lot of exempt dwellings (class `EX`). I wonder if this class restriction needs a little bit more polish?

Reply:
Almost certainly these are thing that _were_ 212s and are now exempt. I would exclude exempt (`EX`) and railroad (`RR`) from this test.

Reply:
Makes sense, I excluded non-regression classes and exempt/RR dwellings in 5fc90ec80f649475291fa47fec80fa543678a208!